### PR TITLE
Front end redesign

### PR DESCRIPTION
**Pull request by Cena-Studio**
# Overview
This PR is part of a redesign of the appearance and UX of the Flight web suite.

# Major Changes
- The UI style of this application is updated to match the new design.
- Add linkage with the console app.

### CODE REVIEW

File: client/src/FileManagerLayout.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      </Overlay>
+    </OverlayContainer>
+  );
+}
+
+function refresh() {
+  document.querySelector('.path-icon.icon-refresh').click();
+}
+
+export default FileManagerLayout;
```
Comment (megan-godfrey) [inline client/src/FileManagerLayout.js:120]: New line at end of each file needed. There are a few other files that need them as well

If it's useful, I often use this bash script to make sure all files have a new line at the end: 

```
#!/bin/bash
for i in $(find ./ -type f \( -iname \*.rb -o -iname \*.html.erb -o -iname \*.js \));
do
  sed -i -e '$a\' $i
done;
```

for loop could be updated to also include css files

File: client/src/FileManagerLayout.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      title="refresh (Ctrl+R)"
+      onClick={refresh}
+    >
+    </i>
+  ) : null;
+
+  const ToConsoleButton = fileManagerState === 'connected' ? (
+    <i
+      title="Open in terminal"
+    >
```
Comment (megan-godfrey) [inline client/src/FileManagerLayout.js:81]: The `<i>` tag isn't needed - you can move the title to the `<a>` tag

File: client/src/FileManagerLayout.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+      <a
+        className="fa fa-terminal ml-2 link white-text"
```
Comment (megan-godfrey) [inline client/src/FileManagerLayout.js:83]: The margins are a bit uneven between the icons. In Console, the middle icon has `ml-2 mr-3`

File: client/src/UnconfiguredDashboard.js [hunk 4]
```diff
@@ -1,33 +1,28 @@
...
+        Flight Console Service allows you to access an interactive terminal
+        console session running on your cluster from the comfort of your
+        browser.
       </p>
 
       <p>
-        Before Flight File Manager Service can be used, it needs to be
-        configured by your system administrator.  It can be configured by
-        running:
+        Before Flight Console Service can be used, it needs to be configured
```
Comment (megan-godfrey) [inline client/src/UnconfiguredDashboard.js:13]: Swap 'Console' for 'Files', here and above :)

I would also remove the word 'Service'. I don't think the apps are referred to in that way anywhere else

File: client/src/UnconfiguredDashboard.js [hunk 5]
```diff
@@ -1,33 +1,28 @@
...
+        by your system administrator.  It can be configured by running:
       </p>
 
       <div className="card card-body">
         <pre className="mb-0">
           <code>
-            flight service configure file-manager-webapp
+            flight service configure console-webapp
```
Comment (megan-godfrey) [inline client/src/UnconfiguredDashboard.js:20]: Revert this change as well

copy paste problems? :stuck_out_tongue:

File: client/src/routes.js [hunk 6]
```diff
@@ -19,33 +19,30 @@ const notFoundRoute = {
...
-    authenticated: true,
-    sideNav: false,
-  },
+  // {
+  //   path: '/browse',
+  //   name: 'My files',
+  //   Component: FileManagerPage,
+  //   authenticated: true,
+  //   sideNav: false,
+  // },
```
Comment (megan-godfrey) [inline client/src/routes.js:32]: Unwanted code to get rid of?

File: client/src/FIleToolbar.js [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+            DOM.CurrentInfo.getData(idFile).then((_, content) => {
+              console.log('content:', content);  // eslint-disable-line no-console
+            });
+          }
+        }}
+        size="sm"
+      >
+        Open in Flight Job
+      </Button>
+      */}
```
Comment (megan-godfrey) [inline client/src/FIleToolbar.js:316]: Unwanted code to remove

File: client/src/FIleToolbar.js [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+
+function NewFileFolderButtons() {
+  const { promptNewDir, promptNewFile } = useContext(FileManagerContext);
+  return (
+    <div className={classNames('d-flex', 'flex-column', 'justify-content-center', 'align-items-center', styles.ButtonGroupWrapper)}>
+      <p className={styles.ButtonGroupTitle}>New</p>
+      <ButtonGroup className={styles.ButtonGroup}>
+        <Button
+          color="light"
+          onClick={promptNewFile}
```
Comment (megan-godfrey) [inline client/src/FIleToolbar.js:101]: A bigger change that isn't super important so don't worry if there's not enough time to change it...

The styling of the modals for naming new files and directories doesn't match the modals from other apps. Here's an example from Flight Desktop...
 
image

If possible it would be good if the modal styling could be consistent

File: client/src/AppLayout.js [hunk 9]
```diff
@@ -18,37 +17,34 @@ function AppLayout() {
...
-function SideNav() {
-  return (
-    <div className="sidenav col-sm-2"></div>
-  );
-}
+// function SideNav() {
+//   return (
+//     <div className="sidenav col-sm-2"></div>
+//   );
+// }
```
Comment (megan-godfrey) [inline client/src/AppLayout.js:48]: Unwanted code to remove

File: client/public/data/index.dev.json [hunk 10]
```diff
@@ -5,7 +5,7 @@
   },
   "apps_link" : {
     "path": "/dev/apps",
-    "text": "Apps"
+    "text": "Web suite"
   },
```
Comment (megan-godfrey) [inline client/public/data/index.dev.json:9]: The `apps_link` field isn't used anymore so lines 6-9 can be removed

File: client/src/FIleToolbar.js [hunk 11]
```diff
@@ -0,0 +1,321 @@
...
+          onClick={pack}
+          size="sm"
+          title="Pack"
+        >
+          <i className="fa fa-compress" />
+        </Button>
+        <Button
+          color="light"
+          // XXX Better disability.
+          // disabled={!isFileSelected}
```
Comment (megan-godfrey) [inline client/src/FIleToolbar.js:272]: Unused code to remove

File: client/src/FileManager.module.css [hunk 12]
```diff
@@ -1,37 +1,57 @@
...
 .FileManager {
     width: 100%;
     height: 100%;
     border: none;
 }
 
-.ButtonGroup :first-child {
-    border-left-color: transparent;
+.FileToolBar {
+    height: 87px;
```
Comment (megan-godfrey) [inline client/src/FileManager.module.css:13]: There are a few issues that arise if the user changes their font size. Here, it would be better to use `rem` rather than `px` so the toolbar height would scale with a user's font size

There are also a few toolbar buttons where the icon goes outside the button in this situation (i.e. the new file, ne directory, rename buttons)

extreme example:
image

File: client/src/FileManager.module.css [hunk 13]
```diff
@@ -1,37 +1,57 @@
...
 }
-.ButtonGroup > *, .ButtonGroup > *:disabled {
-    border-left: 1px solid var(--gray);
-    border-right: 1px solid var(--gray);
+
+.ButtonGroupTitle {
+    margin: 0;
+    color: #3d3d3d;
+    line-height: 16px;
+    font-size: 16px;
```
Comment (megan-godfrey) [inline client/src/FileManager.module.css:32]: As shown in the screenshot above, would be better to define font size in rem so users can increase font size if they need to

**Review by megan-godfrey — CHANGES_REQUESTED** [review]
This all works well and looks good :) I've just mentioned some minor changes that would be good to include.

File: client/src/FileManagerLayout.js (hunk 3 above)
Comment (Cena-Studio) [reply client/src/FileManagerLayout.js:83]: there is no mr-3 in my local Console branch but I've added mr-3 to this one.

File: client/src/FIleToolbar.js (hunk 8 above)
Comment (Cena-Studio) [reply client/src/FIleToolbar.js:101]: Good spot I'm afraid it might not be changed until next time.

File: client/src/FIleToolbar.js (hunk 11 above)
Comment (Cena-Studio) [reply client/src/FIleToolbar.js:272]: Other unused code are removed but this one looks like a legacy comment left by the previous development so I would temporarily keep it for a bit longer.

File: client/src/UnconfiguredDashboard.js [hunk 14]
```diff
@@ -4,20 +4,21 @@ function UnconfiguredDashboard() {
   return (
     <div>
```
Comment (megan-godfrey) [inline client/src/UnconfiguredDashboard.js:5]: Sorry one thing I forgot to check before, the layout of the unconfigured dashboard needs updating - can find an example in flight desktop.

this was something that came up in testing after we stopped working on it

File: client/src/UnconfiguredDashboard.js [hunk 15]
```diff
@@ -4,20 +4,21 @@ function UnconfiguredDashboard() {
...
       <p>
-        Flight Console Service allows you to access an interactive terminal
-        console session running on your cluster from the comfort of your
-        browser.
+        The Flight File Manager Service allows you to manage your files on
```
Comment (megan-godfrey) [inline client/src/UnconfiguredDashboard.js]: Here and in the blurb, the app is being referred to as File Manager, but elsewhere I've changed the app name to Flight Files

**Review by ColonelPanicks — APPROVED** [review]
LGTM! Over to you @VoxSecundus

File: client/.env.development [hunk 16]
```diff
@@ -103,11 +103,11 @@ REACT_APP_FAVICON_URL="/dev/desktop/favicon.ico"
...
-# REACT_APP_BRANDING_CSS_URL="/dev/styles/branding.css"
-# REACT_APP_BRANDING_FILE="/dev/data/branding.json"
-# REACT_APP_ENVIRONMENT_FILE="/dev/data/environment.json"
-# REACT_APP_DATA_FILE="/dev/data/index.json"
-# REACT_APP_FAVICON_URL="/dev/images/favicon.png"
+REACT_APP_BRANDING_CSS_URL="/dev/styles/branding.css"
+REACT_APP_BRANDING_FILE="/dev/data/branding.json"
+REACT_APP_ENVIRONMENT_FILE="/dev/data/environment.json"
+REACT_APP_DATA_FILE="/dev/data/index.json"
+REACT_APP_FAVICON_URL="/dev/images/favicon.png"
```
[the remark this answers is not in the document]
Comment (VoxSecundus) [reply client/.env.development]: I didn't think you'd actually include that :sweat_smile:
